drivers/ec_host_cmd_periph/ec_host_cmd_periph_shi_ite.c

```diff
@@ -0,0 +1,504 @@
+		return -EBUSY;
+	}
+
+	/* Copy preamble */
+	memcpy(data->out_msg, out_preamble, sizeof(out_preamble));
+
+	/* Copy data */
+	memcpy(data->out_msg + sizeof(out_preamble), buf->buf, buf->len);
+
+	/* Append our past-end byte, which we reserved space for. */
+	for (int i = 0; i < EC_SHI_PAST_END_LENGTH; i++) {
+		data->out_msg[sizeof(out_preamble) + buf->len + i] = EC_SHI_PAST_END;
+	}
+
+	tx_size = buf->len + EC_SHI_PREAMBLE_LENGTH + EC_SHI_PAST_END_LENGTH;
+
+	/* Transmit the reply */
+	shi_ite_response_host_data(dev, data->out_msg, tx_size);
+
+	return 0;
+}
+
+/* Store request data from Rx FIFO to in_msg buffer */
+static void shi_ite_host_request_data(uint8_t *in_msg_addr, int count)
+{
+	/* CPU Rx FIFO1 access */
+	IT83XX_SPI_TXRXFAR = IT83XX_SPI_CPURXF1A;
+
+	/*
+	 * In shi_ite_parse_header, the request data will separate to write in_msg buffer so we
+	 * cannot set CPU to end accessing Rx FIFO in this function. We will set
+	 * IT83XX_SPI_TXRXFAR = 0 in shi_ite_reset_rx_fifo.
+	 */
+	for (int i = 0; i < count; i += 4) {
+		/* Get data from master to buffer */
+		*(uint32_t *)(in_msg_addr + i) = IT83XX_SPI_RXFRDRB0;
+	}
+}
+
+static int shi_ite_host_request_expected_size(const struct ec_host_cmd_request_header *r)
+{
+	/* Check host request version */
+	if (r->prtcl_ver != EC_HOST_REQUEST_VERSION) {
+		return 0;
+	}
+
+	/* Reserved byte should be 0 */
+	if (r->reserved) {
+		return 0;
+	}
+
+	return sizeof(*r) + r->data_len;
+}
+
+/* Parse header for version of spi-protocol */
+static void shi_ite_parse_header(const struct device *dev)
+{
+	struct shi_it8xxx2_data *data = dev->data;
+	struct ec_host_cmd_request_header *r = (struct ec_host_cmd_request_header *)data->in_msg;
+
+	/* Store request data from Rx FIFO to in_msg buffer */
+	shi_ite_host_request_data(data->in_msg, sizeof(*r));
+
+	/* Protocol version 3 */
+	if (data->in_msg[0] == EC_HOST_REQUEST_VERSION) {
+		/* Check how big the packet should be */
+		data->in_msg_size = shi_ite_host_request_expected_size(r);
+
+		if (data->in_msg_size == 0 || data->in_msg_size > sizeof(data->in_msg)) {
+			shi_ite_bad_received_data(dev, data->in_msg_size);
+			return;
+		}
+
+		/* Store request data from Rx FIFO to in_msg buffer */
+		shi_ite_host_request_data(data->in_msg + sizeof(*r),
+					  data->in_msg_size - sizeof(*r));
+
+		k_sem_give(&data->handler_owns);
+	} else {
+		/* Invalid version number */
+		LOG_ERR("Invalid version number");
+		shi_ite_bad_received_data(dev, 1);
+	}
+}
+
+static void shi_ite_int_handler(const struct device *dev)
+{
+	struct shi_it8xxx2_data *data = dev->data;
+
+	if (data->shi_state == SHI_STATE_DISABLED) {
+		return;
+	}
+
+	/*
+	 * The status of SPI end detection interrupt bit is set, it means that host command parse
+	 * has been completed and AP has received the last byte which is EC_SHI_PAST_END from
+	 * EC responded data, then AP ended the transaction.
+	 */
+	if (IT83XX_SPI_ISR & IT83XX_SPI_ENDDETECTINT) {
```

Comment:
In this condition, we can use pm_policy_state_lock_put() to make this device enter the power policy to save power. [pm_state=PM_STATE_STANDBY ](https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/riscv/riscv-ite/common/power.c#L25)

Reply:
Added these functions, thank you very much